### PR TITLE
MULE-19107: fix null pointer exception when a header is null

### PR DESCRIPTION
**Pull request by jcgrethe**

### CODE REVIEW

File: modules/http/src/main/java/org/mule/module/http/internal/listener/HttpResponseHeaderBuilder.java [hunk 1]
```diff
@@ -62,6 +62,12 @@ else if (headerValue instanceof String[])
         }
         else
         {
+            if (headerValue == null)
+            {
+                LOGGER.warn("The header " + headerName + " is null. Removing it from the response.");
```
Comment (eze210) [inline modules/http/src/main/java/org/mule/module/http/internal/listener/HttpResponseHeaderBuilder.java:67]: Where is it removed? Is the header removed if you do this?
builder.addHeader("name", "not null");
builder.addHeader("name", null);

File: modules/http/src/test/java/org/mule/module/http/internal/listener/HttpResponseHeaderBuilderTestCase.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
+package org.mule.module.http.internal.listener;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+
+public class HttpResponseHeaderBuilderTestCase {
+
+    @Test
+    public void testNullHeader()
```
Comment (eze210) [inline modules/http/src/test/java/org/mule/module/http/internal/listener/HttpResponseHeaderBuilderTestCase.java:10]: nullHeaderValueIsIgnored()

File: modules/http/src/test/java/org/mule/module/http/internal/listener/HttpResponseHeaderBuilderTestCase.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package org.mule.module.http.internal.listener;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+
+public class HttpResponseHeaderBuilderTestCase {
+
+    @Test
```
Comment (eze210) [inline modules/http/src/test/java/org/mule/module/http/internal/listener/HttpResponseHeaderBuilderTestCase.java:9]: add @Issue("MULE-19017")

**Review by eze210 — CHANGES_REQUESTED** [review]

File: modules/http/src/main/java/org/mule/module/http/internal/listener/HttpResponseHeaderBuilder.java (hunk 1 above)
Comment (jcgrethe) [reply modules/http/src/main/java/org/mule/module/http/internal/listener/HttpResponseHeaderBuilder.java]: Is not remove, the header is not added to the list.

**Review by eze210 — APPROVED** [review]

**Pull request merged**